src/js/actions/GroupsDataActions.js

```diff
@@ -223,44 +227,46 @@ function contains(object, arrayToBeChecked) {
  * @param {object} fileObject - checkdata object.
  * @param {function} dispatch - redux action dispatcher.
  */
-function toggleGroupDataItems(label, fileObject, dispatch) {
+function toggleGroupDataItems(label, fileObject) {
+  let action;
   switch (label) {
     case "comments":
-      dispatch({
+      action = {
         type: consts.TOGGLE_COMMENTS_IN_GROUPDATA,
         contextId: fileObject.contextId,
         text: fileObject.text
-      });
+      };
       break;
     case "reminders":
-      dispatch({
+      action = {
         type: consts.SET_REMINDERS_IN_GROUPDATA,
         contextId: fileObject.contextId,
         boolean: fileObject.enabled
-      });
+      };
       break;
     case "selections":
-      dispatch({
+      action = {
         type: consts.TOGGLE_SELECTIONS_IN_GROUPDATA,
         contextId: fileObject.contextId,
         selections: fileObject.selections
-      });
+      };
       break;
     case "verseEdits":
-      dispatch({
+      action = {
         type: consts.TOGGLE_VERSE_EDITS_IN_GROUPDATA,
         contextId: fileObject.contextId
-      });
+      };
       break;
     case "invalidated":
-      dispatch({
+      action = {
         type: consts.SET_INVALIDATION_IN_GROUPDATA,
         contextId: fileObject.contextId,
         boolean: fileObject.invalidated
-      });
+      };
       break;
     default:
       console.warn("Undefined label in toggleGroupDataItems switch");
       break;
   }
+  return action;
```

Comment:
Action will be `undefined` if the default case is matched.
You should provide a default value for action.